cpu_features: Build error for MinGW

When trying to build GraphBLAS as part of SuiteSparse for MinGW (using MSYS2), compilation fails with the following error:
```
[  7%] Building C object CMakeFiles/graphblas.dir/Source/GB_cpu_features_support.c.obj
In file included from D:/MINGW-packages-mmuetzel/mingw-w64-suitesparse/src/build-MINGW64/GraphBLAS/Source/GB_cpu_features_support.c:16:
D:/MINGW-packages-mmuetzel/mingw-w64-suitesparse/src/build-MINGW64/GraphBLAS/cpu_features/src/hwcaps.c:110:2: error: #error "This platform does not provide hardware capabilities."
  110 | #error "This platform does not provide hardware capabilities."
      |  ^~~~~
D:/MINGW-packages-mmuetzel/mingw-w64-suitesparse/src/build-MINGW64/GraphBLAS/cpu_features/src/hwcaps.c: In function 'GetHardwareCapabilitiesFor':
D:/MINGW-packages-mmuetzel/mingw-w64-suitesparse/src/build-MINGW64/GraphBLAS/cpu_features/src/hwcaps.c:152:26: warning: implicit declaration of function 'GetElfHwcapFromGetauxval'; did you mean 'GetElfHwcapFromProcSelfAuxv'? [-Wimplicit-function-declaration]
  152 |   unsigned long hwcaps = GetElfHwcapFromGetauxval(type);
      |                          ^~~~~~~~~~~~~~~~~~~~~~~~
      |                          GetElfHwcapFromProcSelfAuxv
D:/MINGW-packages-mmuetzel/mingw-w64-suitesparse/src/build-MINGW64/GraphBLAS/cpu_features/src/hwcaps.c: In function 'CpuFeatures_GetPlatformPointer':
D:/MINGW-packages-mmuetzel/mingw-w64-suitesparse/src/build-MINGW64/GraphBLAS/cpu_features/src/hwcaps.c:168:10: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
  168 |   return (const char *)GetHardwareCapabilitiesFor(AT_PLATFORM);
      |          ^
D:/MINGW-packages-mmuetzel/mingw-w64-suitesparse/src/build-MINGW64/GraphBLAS/cpu_features/src/hwcaps.c: In function 'CpuFeatures_GetBasePlatformPointer':
D:/MINGW-packages-mmuetzel/mingw-w64-suitesparse/src/build-MINGW64/GraphBLAS/cpu_features/src/hwcaps.c:172:10: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
  172 |   return (const char *)GetHardwareCapabilitiesFor(AT_BASE_PLATFORM);
      |          ^
make[4]: *** [CMakeFiles/graphblas.dir/build.make:3058: CMakeFiles/graphblas.dir/Source/GB_cpu_features_support.c.obj] Fehler 1
```

Output from the configuration step:
```
( cd GraphBLAS && make )
make[1]: Verzeichnis „/d/MINGW-packages-mmuetzel/mingw-w64-suitesparse/src/build-MINGW64/GraphBLAS“ wird betreten
( cd build && cmake -G"MSYS Makefiles" -DCMAKE_INSTALL_PREFIX="/mingw64" -DCMAKE_BUILD_TYPE=Release -DENABLE_CUDA=OFF -DBLA_VENDOR=OpenBLAS .. && make --jobs=8 )
-- Building SuiteSparse:GraphBLAS version: v7.3.3, date: Dec 1, 2022
-- GraphBLAS C API: v2.0, date: Nov 15, 2021
-- Source:        D:/MINGW-packages-mmuetzel/mingw-w64-suitesparse/src/build-MINGW64/GraphBLAS
-- Build:         D:/MINGW-packages-mmuetzel/mingw-w64-suitesparse/src/build-MINGW64/GraphBLAS/build
-- The C compiler identification is GNU 12.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: C:/msys64/mingw64/bin/gcc.exe - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Install rpath:
-- Build   rpath: D:/MINGW-packages-mmuetzel/mingw-w64-suitesparse/src/build-MINGW64/GraphBLAS/build
-- Build type:    Release
-- Looking for a Fortran compiler
-- Looking for a Fortran compiler - C:/msys64/mingw64/bin/gfortran.exe
-- The Fortran compiler identification is GNU 12.2.0
-- Detecting Fortran compiler ABI info
-- Detecting Fortran compiler ABI info - done
-- Check for working Fortran compiler: C:/msys64/mingw64/bin/gfortran.exe - skipped
-- Fortran: GNU
-- CUDA: not enabled
-- Found OpenMP_C: -fopenmp (found version "4.5")
-- Found OpenMP_Fortran: -fopenmp (found version "4.5")
-- Found OpenMP: TRUE (found version "4.5")
-- cpu_features (by google.com): enabled
-- Building dynamic GraphBLAS library
-- Performing Test TEST_FOR_STDATOMIC
-- Performing Test TEST_FOR_STDATOMIC - Success
-- ANSI C11 atomics: OK. -latomic not needed
-- CMAKE OpenMP libraries:    C:/msys64/mingw64/lib/libgomp.dll.aC:/msys64/mingw64/lib/libmingwthrd.aC:/msys64/mingw64/lib/libmingwthrd.a
-- CMAKE OpenMP include:
-- CMAKE OpenMP C flags:      -fopenmp
-- CMAKE C flags: -march=nocona -msahf -mtune=generic -O2 -pipe -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector-strong -std=c11 -lm -Wno-pragmas  -fexcess-precision=fast  -fcx-limited-range  -fno-math-errno  -fwrapv  -O3 -DNDEBUG -fopenmp
-- Skipping the demos in GraphBLAS/Demo
-- Installation will be system-wide (requires 'sudo make install')
-- ------------------------------------------------------------------------
-- SuiteSparse CMAKE report for: graphblas
-- ------------------------------------------------------------------------
-- install in /mingw64: true
-- install in SuiteSparse/lib and SuiteSparse/include: false
-- build type:           Release
-- NSTATIC:              true (do not build static library)
-- use OpenMP:           yes
-- C compiler:           GNU
-- C flags:              -march=nocona -msahf -mtune=generic -O2 -pipe -Wp,-D_FORTIFY_SOURCE=2 -fstack-protector-strong -std=c11 -lm -Wno-pragmas  -fexcess-precision=fast  -fcx-limited-range  -fno-math-errno  -fwrapv  -O3 -DNDEBUG -fopenmp
-- C++ compiler:
-- C++ flags:
-- C Flags release:      -O3 -DNDEBUG
-- C++ Flags release:
-- compile definitions:
-- ------------------------------------------------------------------------
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    BLA_VENDOR


-- Build files have been written to: D:/MINGW-packages-mmuetzel/mingw-w64-suitesparse/src/build-MINGW64/GraphBLAS/build
```

Building with gcc works fine with the `v733` branch:
```
...
[100%] Linking C shared library libgraphblas.dll
[100%] Built target graphblas
```

Building with clang still fails:
```
[  7%] Building C object CMakeFiles/graphblas.dir/Source/GB_cpu_features_support.c.obj
[  7%] Building C object CMakeFiles/graphblas.dir/Source/GB_cumsum.c.obj
[  7%] Building C object CMakeFiles/graphblas.dir/Source/GB_dealloc_memory.c.obj
In file included from D:/SVN/Octave/GraphBLAS/Source/GB_cpu_features_support.c:16:
D:/SVN/Octave/GraphBLAS/cpu_features/src/hwcaps.c:110:2: error: "This platform does not provide hardware capabilities."
#error "This platform does not provide hardware capabilities."
 ^
D:/SVN/Octave/GraphBLAS/cpu_features/src/hwcaps.c:152:26: warning: call to undeclared function 'GetElfHwcapFromGetauxval'; ISO C99 and later do not
      support implicit function declarations [-Wimplicit-function-declaration]
  unsigned long hwcaps = GetElfHwcapFromGetauxval(type);
                         ^
[  7%] Building C object CMakeFiles/graphblas.dir/Source/GB_debugify_ewise.c.obj
[  8%] Building C object CMakeFiles/graphblas.dir/Source/GB_debugify_mxm.c.obj
1 warning and 1 error generated.
make[2]: *** [CMakeFiles/graphblas.dir/build.make:3058: CMakeFiles/graphblas.dir/Source/GB_cpu_features_support.c.obj] Fehler 1
make[2]: *** Es wird auf noch nicht beendete Prozesse gewartet....
[  8%] Building C object CMakeFiles/graphblas.dir/Source/GB_debugify_reduce.c.obj
make[1]: *** [CMakeFiles/Makefile2:83: CMakeFiles/graphblas.dir/all] Fehler 2
make: *** [Makefile:136: all] Fehler 2
```

--- a/Source/GB_compiler.h
+++ b/Source/GB_compiler.h
@@ -78,22 +78,6 @@
     #define GB_COMPILER_SUB   0
     #define GB_COMPILER_NAME  "IBM xlc " GB_XSTR (__xlC__)
 
-#elif defined ( __MINGW32__ ) || defined ( __MINGW64__ )
-
-    // MinGW (32-bit or 64-bit)
-    #undef  GB_COMPILER_MINGW
-    #define GB_COMPILER_MINGW   1
-
-    #if defined ( __MINGW32__ )
-    #define GB_COMPILER_MAJOR ( __MINGW32_MAJOR_VERSION )
-    #define GB_COMPILER_MINOR ( __MINGW32_MINOR_VERSION )
-    #else
-    #define GB_COMPILER_MAJOR ( __MINGW64_MAJOR_VERSION )
-    #define GB_COMPILER_MINOR ( __MINGW64_MINOR_VERSION )
-    #endif
-    #define GB_COMPILER_SUB   0
-    #define GB_COMPILER_NAME  "MinGW"
-
 #elif defined ( __GNUC__ )
 
     // gcc
@@ -126,6 +110,15 @@
     #define GB_COMPILER_NAME  "other C compiler"
 
 #endif
+
+// MinGW is not a compiler. But a variable with this name is used
+// in different places.
+#if defined ( __MINGW32__ )
+    // MinGW (32-bit or 64-bit)
+    #undef  GB_COMPILER_MINGW
+    #define GB_COMPILER_MINGW   1
+#endif
+
 
 //------------------------------------------------------------------------------
 // Workaround for compiler bug in Microsoft Visual Studio 2019

